scipy/cluster/hierarchy.py

```diff
@@ -629,6 +629,14 @@ def linkage(y, method='single', metric='euclidean'):
         observation vectors; ignored otherwise. See the ``pdist``
         function for a list of valid distance metrics. A custom distance
         function can also be used.
+    optimal_ordering : bool, optional
+        If True, the linkage matrix will be reordering so that the distance
+        between successive leaves is minimal. This results in a more intuitive
+        tree structure when the data are visualized. defaults to False, because
+        this algorithm can be slow, particularly on large datasets [2]_. See also
+        the ``optimal_leaf_ordering`` function.
```

Comment:
I'm assuming you want ``optimal_leaf_ordering`` to be a public function based on this comment. If so, the function should be added to the function listing in the module docstring and the ``__all__`` dict at the top of this file.